CONTRIBUTE.md

```diff
@@ -0,0 +1,177 @@
+comments from a reviewer. The reviewer might help you to identify some
+typos/bugs, to improve your documentation and to improve your
+design/implementation. Don't feel bad about this, this is actually good both for
+you and for the community as a whole! Also, feel free to push back the reviewer
+if you think you have a point. This is a conversation, not a fight.
+
+In order to fix issues spotted by the reviewer, make the additional changes
+locally and commit them on the same feature branch of the pull request.  Do not
+alter commits already submitted to review (e.g. by using rebase), but add your
+changes as new commits. The pull request review UI does not store history of the
+changes, so changing the commits would erase the context that was commented on
+during the review.
+
+1.  Make changes using your favorite editor.
+2.  Run the tests
+3.  git commit -m "Fixing typos spotted by the reviewer" --signoff
+4.  git push origin
+5.  Make changes to the Pull Request message as needed
+    1.  Respond to individual comments as needed
+    2.  Ask the reviewer to take a look (PTAL) in the main pull request thread, as individual comments on changes won't send notifications to the reviewer.
+
+Keep doing this until you receive an Approval from the reviewer.
+
+### 9. Pushing your changes
+After receiving all the approvals needed, the reviewer will pull your changes to
+the repo. Congratulations!
+
+## Pull Request Example
+
+Let's say that you found a bug in which a stack overflow is caused whenever your
+ternary_search function receive a specific set of inputs. You open an issue on
+GitHub detailing your findings and it creates the issue 1234.
+
+To fix this issue you first create a local branch called "bug/1234". You first
+either add a set of regression tests or fix the existing tests to prove that the
+invalid behavior is actually happening. You then decide to commit these changes:
+
+```
+git commit -m "Add regression tests for bug/1234" --sign-off
+(optional) git push origin
+```
+After reading through the code you finally find the source of the issue and fix
+it. You then commit these changes:
+
+```
+git commit -m "Fix algorithm for input X" --sign-off
+(optional) git push origin
+```
+Now both you and the tests are happy with the changes. So you then create a pull
+request from your fork to the base branch. The Pull Request message should be
+detailed, describing the WHY and the HOW of your changes. You then use the
+following message on your Pull Request:
+
+```
+Fix bug in ternary_search that would cause an stack overflow for specific inputs
+
+Whenever three consecutive numbers were used as inputs for the ternary_search
+function it would end up causing a stack overflow. This caused issues whenever
+the user tried to do action X. We fix this by handling these as a special case.
+This closes #1234.
+
+Signed-off-by: Your Name <yourname@yourorg.com>
+```
+Your Pull Request is then assigned to a reviewer. He/she notices that you forgot
```

Comment:
Here (and in other 2 places), use "They" as gender-agnostic pronoun instead of having to write he/she everywhere.